### DISCUSSION TITLE
how to train an LGBMRegressor model on panel data as was done by the winner of the M5 competition

### DISCUSSION THREAD

**Discussion by fkiraly** [alan-turing-institute/sktime#3184] [Q&A]
From discord helpdesk, question by @fisher:

---
hello,
I'm trying to train an LGBMRegressor model on panel data as was done by the winner of the M5 competition [1]. I'm wondering how to do that using sktime. I'm using windowSummarizer [2] to generate lags and rolling features. I also have static and categorical features as well. I'm not sure on how to recursively forecast within the sktime framework. I'm aware of RecursiveTabularRegressionForecaster [3] and the others but I don't know how to use them. Any help would be greatly appreciated. Thanks 

[1]: (https://github.com/Mcompetitions/M5-methods/tree/master/Code%20of%20Winning%20Methods/A1)
[2]: (https://www.sktime.org/en/stable/api_reference/auto_generated/sktime.transformations.series.summarize.WindowSummarizer.html)
[3]: (https://www.sktime.org/en/stable/api_reference/auto_generated/sktime.forecasting.compose.RecursiveTabularRegressionForecaster.html)

**Comment by fkiraly** [comment]
from discord:

---

```python
import pandas as pd
import numpy as np
from sktime.transformations.compose import TransformerPipeline
from sktime.transformations.series.summarize import WindowSummarizer
from sktime.transformations.series.date import DateTimeFeatures
from sktime.forecasting.model_selection import temporal_train_test_split

time = pd.date_range(start='2022-8-3', periods=10)
stores = ['store_a', 'store_b', 'store_c']
index = pd.MultiIndex.from_product([stores, time], names=['store', 'time'])
panel_data = pd.DataFrame(index=index, data=np.random.rand(index.size), columns=['target'])

window_kwargs = {
    "lag_feature": {
        "lag": [1,2,3],
        "mean": [[0, 3]],
        "std": [[0, 3]],
    }
}

pipe = TransformerPipeline(
    steps = [
        (WindowSummarizer(**window_kwargs)),
        (DateTimeFeatures(ts_freq='D')),
    ]
)

features = pipe.fit_transform(panel_data)
static_features = pd.DataFrame(index=stores, data=[['small', 'full menu'], ['large', 'express'], ['small', 'express']], columns=['store_size', 'menu_options'])
all_features = features.merge(static_features, left_on='store', right_index=True)

# all features is X, panel_data is y
# how to use LGBM regressor to do a recursive forecast?

y_train, y_test, X_train, X_test = temporal_train_test_split(panel_data, all_features)
```

**Reply by fkiraly** [reply]
from discord, @fisher:

The above code generate a dataset similar to the one I have
how the data look like

![image](https://user-images.githubusercontent.com/7985502/183065461-c389a9a2-c0c0-4c16-81a8-2f8396fab32e.png)
![image](https://user-images.githubusercontent.com/7985502/183065485-b188360f-42e0-4e4b-944c-aeb1ce1f4300.png)

**Comment by fkiraly** [comment]
FYI @danbartl, @ltsaprounis, @aiwalter - maybe you know the answer? Remember you have also worked on this if I'm not mistaken.

**Comment by fisher** [comment]
@fkiraly I didn't ask that. You can safely close the thread

**Reply by alexanderlange53** [reply]
However, I am very curious about the answer to this question and would appreciate if the thread is kept open
